src/super_gradients/training/utils/checkpoint_utils.py

```diff
@@ -1598,3 +1582,41 @@ def load_pretrained_weights_local(model: torch.nn.Module, architecture: str, pre
 
     pretrained_state_dict = torch.load(pretrained_weights, map_location=map_location)
     _load_weights(architecture, model, pretrained_state_dict)
+    _maybe_load_preprocessing_params(model, pretrained_state_dict)
+
+
+def _load_weights(architecture, model, pretrained_state_dict):
+    if "ema_net" in pretrained_state_dict.keys():
+        pretrained_state_dict["net"] = pretrained_state_dict["ema_net"]
+    solver = YoloXCheckpointSolver() if "yolox" in architecture else DefaultCheckpointSolver()
+    adaptive_load_state_dict(net=model, state_dict=pretrained_state_dict, strict=StrictLoad.NO_KEY_MATCHING, solver=solver)
+    logger.info(f"Successfully loaded pretrained weights for architecture {architecture}")
+
+
+def _maybe_load_preprocessing_params(model: Union[nn.Module, HasPredict], checkpoint: Mapping[str, Tensor]) -> bool:
+    """
+    Tries to load preprocessing params from the checkpoint to the model.
+    The function does not crash, and raises a warning if the loading fails.
+    :param model: Instance of nn.Module
+    :param checkpoint: Entire checkpoint dict (not state_dict with model weights)
+    :return: True if the loading was successful, False otherwise.
+    """
+    model = unwrap_model(model)
+    preprocessing_params_in_checkpoint = "processing_params" in checkpoint.keys()
```

Comment:
`preprocessing_params_in_checkpoint` make it feel like it represents the preprocessing params, when it is in fact a bool.
`checkpoint_has_preprocessing_params` or `is_checkpoint_with_preprocessing_params` could be clearer